GO connection to  Port Documentation (#354)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/go.md

---
sidebar_position: 12
description: Ingest GO Modules and Dependencies into your catalog
---

import GoBlueprint from "./resources/go/\_example_go_blueprint.mdx";
import GoConfiguration from "./resources/go/\_example_go_webhook_configuration.mdx"
import GoShell from "./resources/go/\_example_go_bash.mdx"

# GO

In this example you are going to create a webhook integration between [GO](https://go.dev/) and Port, which will ingest GO Modules,Version and Dependencies to Port and map them to your Blueprint entities. This integration will involve setting up a webhook to keep track of module,version and dependencies of your Golang application whenever a go lang application is created or updated, allowing Port to ingest and process the GO entities accordingly.

## Prerequisites

> Create the following blueprint definitions and webhook configuration:

Reviewer: Add a :::note block specifying the different unique commands the script will use (I noticed jq and mapfile)

<details>
<summary>GO module blueprint</summary>
<GoBlueprint/>
</details>

<details>
<summary>GO webhook configuration</summary>
<GoConfiguration/>
</details>

<details>
<summary>GO Bash Script</summary>
<GoShell/>
</details>

## Parsing go.mod file and sending dependency data to Port

### Script Usage

1. Copy the script into a file in the root of your Go project. Make sure your go.mod file is also located in the root of the project;

> 2. Make the script executable. For instance, if you named the script parse_and_send.sh, you would use the following command;

Reviewer: ```suggestion
2. Make the script executable. For instance, if you named the script `parse_and_send.sh`, use the following command:
```

   ```bash showLineNumbers
   chmod +x parse_and_send.sh
   ```
3. Run the script:
   ```bash showLineNumbers
   ./parse_and_send.sh
   ```

> Done! After the script has run, it will automatically injest Go packages into Port via HTTP Request

Reviewer: ```suggestion
Done! After the script completes, You will see Go dependency entities in Port.
```

---
sidebar_position: 12
description: Ingest GO Modules and Dependencies into your catalog
---

import GoBlueprint from "./resources/go/\_example_go_blueprint.mdx";
import GoConfiguration from "./resources/go/\_example_go_webhook_configuration.mdx"
import GoShell from "./resources/go/\_example_go_bash.mdx"

# GO

In this example you are going to create a webhook integration between [GO](https://go.dev/) and Port, which will ingest GO Modules,Version and Dependencies to Port and map them to your Blueprint entities. This integration will involve setting up a webhook to keep track of module,version and dependencies of your Golang application whenever a go lang application is created or updated, allowing Port to ingest and process the GO entities accordingly.

## Prerequisites

Create the following blueprint definitions and webhook configuration:

<details>
<summary>GO module blueprint</summary>
<GoBlueprint/>
</details>

<details>
<summary>GO webhook configuration</summary>
<GoConfiguration/>
</details>

<details>

> <summary>GO Bash configuration</summary>

Reviewer 2: Go Bash Script*

Author: This is corrected in my next commit

<GoShell/>
</details>

## Parsing go.mod file and sending dependency data to Port

### Script Usage
1. Copy the script into a file in the root of your Go project. Make sure your go.mod file is also located in the root of the project;
2. Make the script executable. For instance, if you named the script parse_and_send.sh, you would use the following command;

>     ```json showLineNumbers

Reviewer 2: why json here?

Author: This is corrected

    chmod +x parse_and_send.sh
    ```
3. Run the script:

>     ```json showLineNumbers

Reviewer 2: also here

    ./parse_and_send.sh
    ```

> Done! After the script has run, you should see a new file in your project root named output.json containing the parsed go.mod data. The script also prints the HTTP response code after attempting to send the data to Port.

Reviewer 2: Don't explain about the output json, maybe explain it in a tip

Here just write that it autootically ingests it to Port via HTTP request

Author: This change is completed

---
sidebar_position: 12

> description: Ingest Go Modules and dependencies into your catalog

Reviewer: ```suggestion
description: Ingest Go modules and dependencies into your catalog
```

---

import GoBlueprint from "./resources/go/\_example_go_blueprint.mdx";
import GoConfiguration from "./resources/go/\_example_go_webhook_configuration.mdx"
import GoShell from "./resources/go/\_example_go_bash.mdx"

# Go

In this example you will create a webhook integration that will injest [Go](https://go.dev/) modules, versions and dependencies to Port.

The integration includes setting up a webhook to keep track of the dependencies defined in your Golang application, allowing Port to ingest and process the Go entities accordingly.

## Prerequisites

Create the following blueprint definitions and webhook configuration:

<details>
<summary>Go module blueprint</summary>
<GoBlueprint/>
</details>

<details>
<summary>Go webhook configuration</summary>
<GoConfiguration/>
</details>

<details>

> <summary>Go Bash Script</summary>

Reviewer: ```suggestion
<summary>Go bash script</summary>
```

   ./parse_and_send.sh
   ```

:::note

> - `mapfile -t` reads lines into an array from the `go.mod` file, creating a collection for iteration.

Reviewer: It is not necessary to explain what the commands do, just to mention commands that might not be available in the user's shell by default and will require installation (sed, tail, head etc. are always available, but `jq` might not be on some shells so it is worth mentioning is. I noticed that `mapfile` is also unavailable for me so it is also worth mentioning)
